docs/reference/evaluate_pipeline.md

```diff
@@ -1,151 +1,153 @@
-# The evaluate function
+# The `evaluate_pipeline` function
 
-## Introduction
+> **TL;DR**
+> *Sync*: return a scalar or a dict ⟶ NePS records it automatically.
+> *Async*: return `None`, launch a job, and call `neps.save_pipeline_results()` when the job finishes.
 
-The `evaluate_pipeline=` function is crucial for NePS. It encapsulates the objective function to be minimized, which could range from a regular equation to a full training and evaluation pipeline for a neural network.
+---
 
-This function receives the configuration to be utilized from the parameters defined in the search space. Consequently, it executes the same set of instructions or equations based on the provided configuration to minimize the objective function.
+## 1  Return types
 
-We will show some basic usages and some functionalites this function would require for successful implementation.
+| Allowed return | When to use                                 | Minimal example                              |
+| -------------- | ------------------------------------------- | -------------------------------------------- |
+| **Scalar**     | simple objective, single fidelity           | `return loss`                                |
+| **Dict**       | need cost/extra metrics                     | `{"objective_to_minimize": loss, "cost": 3}` |
+| **`None`**     | you launch the job elsewhere (SLURM, k8s …) | *see § 3 Async*                              |
 
-## Types of Returns
+All other values raise a `TypeError` inside NePS.
 
-### 1. Single Value
+## 2  Result dictionary keys
 
-Assuming the `pipeline_space=` was already created (have a look at [pipeline space](./pipeline_space.md) for more details).
-A `evaluate_pipeline=` function with an objective of minimizing the loss will resemble the following:
+| key                     | purpose                                                                      | required?                     |
+| ----------------------- | ---------------------------------------------------------------------------- | ----------------------------- |
+| `objective_to_minimize` | scalar NePS will minimise                                                    | **yes**                       |
+| `cost`                  | wall‑clock, GPU‑hours, … — only if you passed `max_cost_total` to `neps.run` | yes *iff* cost budget enabled |
+| `learning_curve`        | list/np.array of intermediate objectives                                     | optional                      |
```

Comment:
could this instead be a `dict` or `pd.Series/DataFrame`?
if we are creating an explicit *learning_curve* key, then we should also allow for more information to be passed, such as the step count or FLOPs, etc.

alternatively, we drop this key altogether, keep things lean, and let the `extra` column take care of things